### ISSUE TITLE
Opening a modal or dropdown from a modal breaks keyboard tabindex for the whole page

### ISSUE THREAD

**Issue by eatyourgreens** [zooniverse/front-end-monorepo#2183]
## Package
app-project
lib-classifier

## Describe the bug
On a page with previously transcribed lines, click on a purple line and complete it by filling out the subtask box. After closing the subtask, all focusable elements in the classifier have `tabindex=-1`, as if disabled. This includes the image toolbar buttons and radio inputs in the task area.

## To Reproduce
1. Go to https://frontend.preview.zooniverse.org/projects/blicksam/transcription-task-testing/classify/workflow/13898
2. Click on a purple line (you might need to refresh to get a subject which isn't 100% complete.)
3. Choose an option from the dropdown and close the subtask box.
4. Try navigating to any other part of the classifier with Tab.

## Additional context
I've been able to reproduce this bug, in the dev classifier, as far back as commit 858ab002.

**Comment by eatyourgreens** [comment]
Here's the relevant code in Grommet. In our case, `TABINDEX_STATE` is -1, so remains set.

https://github.com/grommet/grommet/blob/8a8ef34f119bf6dd3b5768436257d203e7c7f5fa/src/js/utils/DOM.js#L117-L121

**Comment by eatyourgreens** [comment]
You can also trigger this bug by opening the confirmation dialog on top of an incomplete subtask. The second modal layer remembers the state of the DOM when it opens, which is disabled, then restores that state when it closes.

**Comment by eatyourgreens** [comment]
I can also trigger this bug by adding to a collection. The dropdown suggestions box updates the `data-g-tabindex` attribute from `none` to '-1'.

**Comment by eatyourgreens** [comment]
Opened as https://github.com/grommet/grommet/issues/5260 for the `TextInput` with suggestions.

**Comment by eatyourgreens** [comment]
Updated title because tabindex is set across the entire Classify page, not just the Classifier component.

**Comment by eatyourgreens** [comment]
#2184 will fix this for subtask confirmation popups but I'm not sure how we want to handle navigating the classifier when two dialogs are open. Maybe having two dialogs open at the same time should be avoided, from a UX or accessibility point of view?

**Comment by srallen** [comment]
Generally, yes it probably should be avoided, but I think the specific case of the sub-task and confirmation is appropriate and needed.

**Comment by eatyourgreens** [comment]
We should be passing `trapFocus: false` to `Drop` when it’s used inside a modal layer. See https://github.com/grommet/grommet/pull/4378

**Comment by eatyourgreens** [comment]
Based on the work I’ve been doing in #2198,`ConfirmModal` should be required (it should not be possible to dismiss it without making a decision as to whether you intend to continue the incomplete task.) The task popup itself should be completely disabled while awaiting confirmation.

I’m also wondering why we allow `SubtaskPopup` to be dismissed, either by pressing Esc or clicking outside the popup, for required tasks. That seems counterintuitive.
